src/constraint/spec.ts

```diff
@@ -532,9 +554,30 @@ export const SPEC_CONSTRAINTS: SpecConstraintModel[] = [
     }
   },
   {
+     name: 'scaleBandSizeMustMatchScaleType',
```

Comment:
Our last conversation, we mentiont that we only want one `scalePropertiesSupportedByScaleType` rather than having one for each scale property.   Please jot down something we already agree on.  It's not fun to have to keep repeating something for you.   

Plus, why is this a spec constraint?  (It seems like an encoding constraint for me.)